windows/hardening/AD-Hardening.ps1

```diff
@@ -446,6 +468,254 @@ function Configure-Secure-GPO {
     }
 }
 
+function Create-Good-GPO {
+    try {
+        Write-Host "Creating GPO named '$GPOName'..." -ForegroundColor Green
+        $newGPO = New-GPO -Name $GPOName
+
+        Write-Host "Fetching distinguished name of the domain..." -ForegroundColor Green
+        $domainDN = (Get-ADDomain).DistinguishedName
+
+        Write-Host "Linking GPO to the domain..." -ForegroundColor Green
+        New-GPLink -Name $GPOName -Target $domainDN
+		Write-Host "GPO linked successfully!" -ForegroundColor Green
+
+        Write-Host "Setting permissions for GPO..." -ForegroundColor Green
+        # Get the SID of the current user
+        $userSID = (New-Object System.Security.Principal.NTAccount($CurrentUser)).Translate([System.Security.Principal.SecurityIdentifier]).Value
+
+        # Set permissions for the creating user (full control)
+		try {
+			Set-GPPermissions -Name $GPOName -TargetName $CurrentUser -TargetType User -PermissionLevel GpoEdit
+			Write-Host "Permissions set successfully." -ForegroundColor Green
+		} catch {
+			Write-Host "Error setting permissions -- $_" -ForegroundColor Yellow
+		}
+		Write-Host "Go configure the GPO, specifically to deny the 'Apply Group Policy' for current user, before continuing" -ForegroundColor Black -BackgroundColor Yellow
+		Read-Host " "
+		Set-GPLink -Name $GPOName -Target $domainDN -Enforced Yes
+		Write-Host "GPO fully and successfully configured and enforced!" -ForegroundColor Green
+    } catch {
+        Write-Host $_.Exception.Message -ForegroundColor Yellow
+        Write-Host "Error Occurred..."
+    }
+}
+
+function Configure-Secure-GPO {
+    try {
+        Handle-First-Policy-in-GPO
+
+        # Define configurations
+        $configurations = @{
```

Comment:
This looks like a good candidate for a JSON file instead of updating the code every time.

Reply:
True, I'm not doing that for now, especially because this code is disabled regardless.